process/block/baseProcess.go

```diff
@@ -654,7 +692,13 @@ func (bp *baseProcessor) setProcessingTypeAndConstructionStateForScheduledMb(
 			return err
 		}
 	} else {
-		err = miniBlockHeaderHandler.SetConstructionState(int32(block.Final))
+		constructionState := int32(block.Final)
+		processedMiniBlockInfo := processedMiniBlocksDestMeInfo[string(miniBlockHeaderHandler.GetHash())]
+		if processedMiniBlockInfo != nil && !processedMiniBlockInfo.IsFullyProcessed {
```

Comment:
this conditional can be extracted into a function IsPartiallyExecuted

Reply:
Done